src/app/pages/view-test-result/components/faults-data-row/faults-data-row.ts

```diff
@@ -50,8 +50,12 @@ export class FaultsDataRowComponent {
    * OR fault comments exist when no serious/dangerous faults exist
    * @param drivingFault
    */
-  showFaultComment = (drivingFault: FaultSummary): boolean =>
-    this.drivingFaultCount > this.minDrivingFaultCount && drivingFault.comment !== null;
+  showFaultComment(drivingFault: FaultSummary): boolean {
+    if (drivingFault.comment === (undefined || '')) {
+      drivingFault.comment = null;
+    }
+    return this.drivingFaultCount > this.minDrivingFaultCount && drivingFault.comment !== null;
+  }
```

Comment:
This can be simplified

`
showFaultComment = (drivingFault: FaultSummary): boolean => this.drivingFaultCount > this.minDrivingFaultCount && !!(drivingFault.comment);
`